src/Cache/MongoStore.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+namespace MongoDB\Laravel\Cache;
+
+use Illuminate\Cache\RetrievesMultipleKeys;
+use Illuminate\Contracts\Cache\LockProvider;
+use Illuminate\Contracts\Cache\Store;
+use Illuminate\Support\InteractsWithTime;
+use MongoDB\Laravel\Collection;
+use MongoDB\Laravel\Connection;
+use MongoDB\Operation\FindOneAndUpdate;
+use Override;
+
+use function is_float;
+use function is_int;
+use function is_string;
+use function serialize;
+use function str_contains;
+use function unserialize;
+
+final class MongoStore implements LockProvider, Store
+{
+    use InteractsWithTime;
+    // Provides "many" and "putMany" in a non-optimized way
+    use RetrievesMultipleKeys;
+
+    private Collection $collection;
+
+    public function __construct(
+        private Connection $connection,
+        private string $collectionName,
+        private string $prefix = '',
+        private ?Connection $lockConnection = null,
+        private string $lockCollectionName = 'cache_locks',
+        private array $lockLottery = [2, 100],
+        private int $defaultLockTimeoutInSeconds = 86400,
+    ) {
+        $this->collection = $this->connection->getCollection($this->collectionName);
+    }
+
+    /**
+     * Get a lock instance.
+     *
+     * @param string      $name
+     * @param int         $seconds
+     * @param string|null $owner
+     */
+    #[Override]
+    public function lock($name, $seconds = 0, $owner = null): MongoLock
+    {
+        return new MongoLock(
+            ($this->lockConnection ?? $this->connection)->getCollection($this->lockCollectionName),
+            $this->prefix . $name,
+            $seconds,
+            $owner,
+            $this->lockLottery,
+            $this->defaultLockTimeoutInSeconds,
+        );
+    }
+
+    /**
+     * Restore a lock instance using the owner identifier.
```

Comment:
I realize this was copied from [DatabaseStore](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Cache/DatabaseStore.php#L295), but how is this method supposed to be used? I don't see it in the test suite.

"Restore a lock instance using the owner identifier" seems like an inaccurate description for a method that constructs and returns a new MongoLock instance.

I assume this has something to do with the `$owner` being a required parameter, which will differ from `lock()` generating a random string by default.

Reply:
Apparently, that's useful for cross-process locks: https://laravel.com/docs/11.x/cache#managing-locks-across-processes

It's part of the `LockProvider` interface, so I must implement it. But all the implementations are identical.